plugins/woocommerce/src/Internal/Fulfillments/OrderFulfillmentsRestController.php

```diff
@@ -0,0 +1,1084 @@
+						'description' => __( 'The tracking URL.', 'woocommerce' ),
+						'type'        => 'string',
+						'required'    => true,
+					),
+				),
+			),
+		);
+		return $schema;
+	}
+
+	/**
+	 * Get the base schema for the fulfillment with a read context.
+	 *
+	 * @return array
+	 */
+	private function get_read_schema_for_fulfillment() {
+		return array(
+			'fulfillment_id' => array(
+				'description' => __( 'Unique identifier for the fulfillment.', 'woocommerce' ),
+				'type'        => 'integer',
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+			),
+			'entity_type'    => array(
+				'description' => __( 'The type of entity for which the fulfillment is created.', 'woocommerce' ),
+				'type'        => 'string',
+				'required'    => true,
+				'context'     => array( 'view', 'edit' ),
+			),
+			'entity_id'      => array(
+				'description' => __( 'Unique identifier for the entity.', 'woocommerce' ),
+				'type'        => 'string',
+				'required'    => true,
+				'context'     => array( 'view', 'edit' ),
+			),
+			'status'         => array(
+				'description' => __( 'The status of the fulfillment.', 'woocommerce' ),
+				'type'        => 'string',
+				'default'     => 'unfulfilled',
+				'required'    => true,
+				'context'     => array( 'view', 'edit' ),
+			),
+			'is_fulfilled'   => array(
+				'description' => __( 'Whether the fulfillment is fulfilled.', 'woocommerce' ),
+				'type'        => 'boolean',
+				'default'     => false,
+				'required'    => true,
+				'context'     => array( 'view', 'edit' ),
+			),
+			'date_updated'   => array(
+				'description' => __( 'The date the fulfillment was created.', 'woocommerce' ),
+				'type'        => 'string',
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+				'required'    => true,
+			),
+			'date_deleted'   => array(
+				'description' => __( 'The date the fulfillment was deleted.', 'woocommerce' ),
+				'type'        => 'string',
+				'default'     => null,
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+				'required'    => true,
+			),
+			'meta_data'      => array(
+				'description' => __( 'Meta data for the fulfillment.', 'woocommerce' ),
+				'type'        => 'array',
+				'required'    => true,
+				'schema'      => $this->get_schema_for_meta_data(),
+			),
+		);
+	}
+
+	/**
+	 * Get the base args for the fulfillment with a write context.
+	 *
+	 * @param bool $is_create Whether the args list is for a create request.
+	 *
+	 * @return array
+	 */
+	private function get_write_args_for_fulfillment( bool $is_create = false ) {
+		return array_merge(
+			! $is_create ? array(
+				'fulfillment_id' => array(
+					'description' => __( 'Unique identifier for the fulfillment.', 'woocommerce' ),
+					'type'        => 'integer',
+					'context'     => array( 'view', 'edit' ),
+					'readonly'    => true,
+				),
+			) : array(),
+			array(
+				'status'       => array(
```

Comment:
Same as https://github.com/woocommerce/woocommerce/pull/57496/files#r2066478301

Reply:
Possible values for that will be dynamic, as the user will be able to add more with a filter provided in a separate PR. I will have to change that if I moved that into an enum when working on that future PR.